plugin/plugin_scaffold.go

```diff
@@ -0,0 +1,169 @@
+package plugin
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"unicode"
+
+	"github.com/cybercyst/go-scaffold/pkg/scaffold"
+	gerr "github.com/gatewayd-io/gatewayd/errors"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	FolderPermissions os.FileMode = 0o755
+	FilePermissions   os.FileMode = 0o644
+)
+
+// Scaffold generates a gatewayd plugin based on the provided input file
+// and stores the generated files in the specified output directory.
+// The expected input file should be a YAML file containing the following fields:
+//
+// ```yaml
+// remote_url: https://github.com/gatewayd-io/test-gatewayd-plugin
+// version: 0.1
+// description: This is test plugin
+// license: MIT
+// authors:
+//   - GatewayD Team
+func Scaffold(inputFile string, outputDir string) ([]string, error) {
+	tempDir, err := os.MkdirTemp("", "gatewayd-plugin-template")
+	if err != nil {
+		return nil, gerr.ErrFailedToScaffoldPlugin.Wrap(err)
+	}
+
+	defer func() {
+		os.RemoveAll(tempDir)
+	}()
+
+	// Copy the embedded directory and its contents as "go-scaffold" library only accepts files on os filesystem
+	err = fs.WalkDir(pluginTemplate, pluginTemplateRootDir, func(path string, dir fs.DirEntry, err error) error {
+		if err != nil {
+			return gerr.ErrFailedToCopyEmbeddedFiles.Wrap(err)
+		}
+
+		relativePath, err := filepath.Rel(pluginTemplateRootDir, path)
+		if err != nil {
+			return gerr.ErrFailedToCopyEmbeddedFiles.Wrap(err)
+		}
+		destPath := filepath.Join(tempDir, relativePath)
+
+		if dir.IsDir() {
+			return os.MkdirAll(destPath, FolderPermissions)
+		}
+
+		fileContent, err := pluginTemplate.ReadFile(path)
+		if err != nil {
+			return gerr.ErrFailedToCopyEmbeddedFiles.Wrap(err)
+		}
+
+		if err := os.MkdirAll(filepath.Dir(destPath), FolderPermissions); err != nil {
+			return gerr.ErrFailedToCopyEmbeddedFiles.Wrap(err)
+		}
+
+		return os.WriteFile(destPath, fileContent, FilePermissions)
+	})
+	if err != nil {
+		return nil, gerr.ErrFailedToScaffoldPlugin.Wrap(err)
+	}
+
+	input, err := readScaffoldInputFile(inputFile)
+	if err != nil {
+		return nil, gerr.ErrFailedToScaffoldPlugin.Wrap(err)
+	}
+
+	var pluginName string
+	if url, ok := input["remote_url"].(string); ok {
+		// Type assertion succeeded, url is now a string
```

Comment:
```suggestion
```

Reply:
Fixed in https://github.com/gatewayd-io/gatewayd/pull/518/commits/782795ac25b2f8bf16c9b363111518c25eef1309.